r2-netty/src/main/java/com/linkedin/r2/netty/client/http2/Http2ChannelLifecycle.java

```diff
@@ -221,7 +221,7 @@ public void onSuccess(Channel channel)
           }
         });
       }
-    }, new ObjectCreationTimeoutException(
+    }, () -> new ObjectCreationTimeoutException(
     "Exceeded creation timeout of " + _channelCreationTimeoutMs + "ms: for HTTP/2 parent channel, remote=" + _address)));
```

Comment:
Now instead of creating a throwable object, we are creating a lambda. How can we be sure that the cost of a lambda is smaller than the cost of the throwable object?

Reply:
See pervious comments.

Reply:
I understand that we are trying to lazily create the object instance. But my question is more on the cost of creating a lambda. Since the lambda also involves a closure, wouldnt that create a seperate lambda object for each request. So the question is, did we validate if the cost of lambda object for each request is smaller than the cost of creating the Throwable itself?

Reply:
According to this (https://stackoverflow.com/questions/2347828/how-expensive-is-thread-getstacktrace/10625065#10625065), the answer is definitely yet.